packages/teleport/src/Discover/SelectResource/SelectResource.test.tsx

```diff
@@ -0,0 +1,337 @@
+
+      const proceedButton = result.getByTestId('proceed-button');
+
+      expect(proceedButton.hasAttribute('disabled')).toBe(true);
+    });
+
+    test('has the proceed button enabled when having correct permissions', () => {
+      const result = create('database', fullAcl);
+
+      expect(result.container.firstChild).toMatchSnapshot();
+
+      const proceedButton = result.getByTestId('proceed-button');
+
+      expect(proceedButton.hasAttribute('disabled')).toBe(false);
+    });
+  });
+
+  describe('kubernetes', () => {
+    test('shows permissions error when lacking tokens.create', () => {
+      const result = create('kubernetes', {
+        ...fullAcl,
+        tokens: {
+          ...fullAccess,
+          create: false,
+        },
+      });
+
+      const permissionsError = result.getByTestId('permissions-error');
+      expect(permissionsError).toBeInTheDocument();
+
+      expect(result.container.firstChild).toMatchSnapshot();
+
+      const proceedButton = result.getByTestId('proceed-button');
+
+      expect(proceedButton.hasAttribute('disabled')).toBe(true);
+    });
+
+    test('shows permissions error when lacking kubeServers.read', () => {
+      const result = create('kubernetes', {
+        ...fullAcl,
+        kubeServers: {
+          ...fullAccess,
+          read: false,
+        },
+      });
+
+      const permissionsError = result.getByTestId('permissions-error');
+      expect(permissionsError).toBeInTheDocument();
+
+      expect(result.container.firstChild).toMatchSnapshot();
+
+      const proceedButton = result.getByTestId('proceed-button');
+
+      expect(proceedButton.hasAttribute('disabled')).toBe(true);
+    });
+
+    test('has the proceed button enabled when having correct permissions', () => {
+      const result = create('kubernetes', fullAcl);
+
+      expect(result.container.firstChild).toMatchSnapshot();
+
+      const proceedButton = result.getByTestId('proceed-button');
+
+      expect(proceedButton.hasAttribute('disabled')).toBe(false);
+    });
+  });
+
+  describe('application', () => {
+    test('shows permissions error when lacking tokens.create', () => {
+      const result = create('application', {
+        ...fullAcl,
+        tokens: {
+          ...fullAccess,
+          create: false,
+        },
+      });
+
+      const permissionsError = result.getByTestId('permissions-error');
+      expect(permissionsError).toBeInTheDocument();
+
+      expect(result.container.firstChild).toMatchSnapshot();
+
+      const proceedButton = result.getByTestId('proceed-button');
+
+      expect(proceedButton.hasAttribute('disabled')).toBe(true);
+    });
+
+    test('shows permissions error when lacking appServers.read', () => {
+      const result = create('application', {
+        ...fullAcl,
+        appServers: {
+          ...fullAccess,
+          read: false,
+        },
+      });
+
+      const permissionsError = result.getByTestId('permissions-error');
```

Comment:
[nonblocking]
Instead of using test ids try using `getByText` here.
https://testing-library.com/docs/queries/about/#priority

Reply:
full eg:
```
    test('shows permissions error when lacking appServers.read', () => {
      const result = create('application', {
        ...fullAcl,
        appServers: {
          ...fullAccess,
          read: false,
        },
      });

      expect(result.getByText(/You are not able to create./)).toBeInTheDocument();
      expect(result.getByText(/Your Teleport Enterprise license does not include application/)).toBeInTheDocument();
      expect(result.getByRole('button', { name: "Next" })).toBeDisabled();
    });
```